storybookjs/storybook
# Move 'dev'/'build' events to the end of the process

Issue: N/A

## What I did

Move the point where we call out to `'dev'` and `'build'` to be *after* we know it has "worked".

## How to test

1. Start the event log collector (`cd scripts; yarn ts-node ./event-log-collector.ts`)
2. Create a sandbox
3. Run dev + build, check they log `boot` and `dev`/`build`.
4. Add an error to `preview.js` (e.g. `const = 7;`)
5. Run dev + build, check they log `boot` and `error`.

Note the above won't fail in Vite dev mode, you'd have to be more creative.

Also note, only part 3 for `build` is current tested in CI, would be good to figure out how to change that.
